python/common/org/python/types/Bytes.java

```diff
@@ -971,10 +972,56 @@ public boolean __setattr_null(java.lang.String name, org.python.Object value) {
     }
 
     @org.python.Method(
-            __doc__ = "B.expandtabs(tabsize=8) -> copy of B\n\nReturn a copy of B where all tab characters are expanded using spaces.\nIf tabsize is not given, a tab size of 8 characters is assumed."
+            __doc__ = "B.expandtabs(tabsize=8) -> copy of B\n\nReturn a copy of B where all tab characters are expanded using spaces.\nIf tabsize is not given, a tab size of 8 characters is assumed.",
+            default_args = {"tabsize"}
     )
-    public org.python.Object expandtabs(java.util.List<org.python.Object> args, java.util.Map<java.lang.String, org.python.Object> kwargs, java.util.List<org.python.Object> default_args, java.util.Map<java.lang.String, org.python.Object> default_kwargs) {
-        throw new org.python.exceptions.NotImplementedError("bytes.expandtabs has not been implemented.");
+    public org.python.Object expandtabs(org.python.types.Object tabsize) {
+        // CPython implementation https://github.com/python/cpython/blob/3.6/Objects/stringlib/transmogrify.h#L21
+        // TODO: check overflows
+
+        if (this.value == null) {
+            return null;
+        }
+
+        if (this.value.length == 0) {
+            return new org.python.types.Bytes(new byte[0]);
+        }
+
+        int tabsizeInt = 8;
+
+        if (tabsize != null) {
+            tabsizeInt = (int) ((org.python.types.Int) tabsize).value;
+        }
+
+        ArrayList<Character> buffer = new ArrayList<>();
```

Comment:
This should probably be of type `Byte` instead of `Character` (one Character uses two bytes in UTF-16 -- the default Java charset).
You could also pass an initial capacity here to `ArrayList<>(this.values.length)` to reduce the number of reallocations needed.